tests/integration/helpers/cluster.py

```diff
@@ -563,26 +598,33 @@ def cleanup(self):
                     logging.debug(f"Unstopped containers killed.")
             else:
                 logging.debug(f"No running containers for project: {self.project_name}")
-        except:
-            pass
+        except Exception as ex:
+            logging.debug(f"Got exception removing containers {str(ex)}")
 
         # # Just in case remove unused networks
-        # try:
-        #     logging.debug("Trying to prune unused networks...")
-
-        #     run_and_check(['docker', 'network', 'prune', '-f'])
-        #     logging.debug("Networks pruned")
-        # except:
-        #     pass
+        try:
+            logging.debug("Trying to prune unused networks...")
+
+            list_networks = subprocess.check_output(
+                f"docker network ls -q --filter name='{self.project_name}'",
+                shell=True,
+                universal_newlines=True,
+            ).splitlines()
+            if list_networks:
+                logging.debug(f"Trying to remove networks: {list_networks}")
+                run_and_check(f"docker network rm {' '.join(list_networks)}")
+                logging.debug(f"Networks removed: {list_networks}")
+        except:
+            pass
 
         # Remove unused images
-        # try:
-        #     logging.debug("Trying to prune unused images...")
+        try:
+            logging.debug("Trying to prune unused images...")
 
-        #     run_and_check(['docker', 'image', 'prune', '-f'])
-        #     logging.debug("Images pruned")
-        # except:
-        #     pass
+            run_and_check(["docker", "image", "prune", "-f"])
+            logging.debug("Images pruned")
```

Comment:
You just forgot what prune does.
Only those images that are outdated and newer version is also present will be pruned. It will not remove even old versions for `clickhouse-server` with specified tags. But will remove old images that previously also had `latest` tag or similar.
To remove unused images `-a` should be defined and it is not.